### PR TITLE
fix(client): import TFunction from i18next in profile componenets

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Ref #49800

TFunction is local type in react-i18next, so this aims to import it from where it's valid

### CODE REVIEW

File: client/src/components/layouts/default.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React, { ReactNode, useEffect } from 'react';
 import Helmet from 'react-helmet';
-import { TFunction, withTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
```
Comment (Sboonny) [inline client/src/components/layouts/default.tsx:3]:
```suggestion
import type { TFunction } from 'i18next';
```

File: client/src/components/profile/components/camper.tsx [hunk 2]
```diff
@@ -6,7 +6,8 @@ import {
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { Col, Row } from '@freecodecamp/react-bootstrap';
 import React from 'react';
-import { TFunction, useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
```
Comment (Sboonny) [inline client/src/components/profile/components/camper.tsx:9]:
```suggestion
import type { TFunction } from 'i18next';
```

File: client/src/components/profile/components/heat-map.tsx [hunk 3]
```diff
@@ -7,7 +7,8 @@ import addMonths from 'date-fns/addMonths';
 import isEqual from 'date-fns/isEqual';
 import startOfDay from 'date-fns/startOfDay';
 import React, { Component } from 'react';
-import { TFunction, useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
```
Comment (Sboonny) [inline client/src/components/profile/components/heat-map.tsx:10]:
```suggestion
import type { TFunction } from 'i18next';
```

File: client/src/components/profile/components/time-line.tsx [hunk 4]
```diff
@@ -3,7 +3,8 @@ import Loadable from '@loadable/component';
 import { graphql, useStaticQuery } from 'gatsby';
 import { reverse, sortBy } from 'lodash-es';
 import React, { useMemo, useState } from 'react';
-import { TFunction, withTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
```
Comment (Sboonny) [inline client/src/components/profile/components/time-line.tsx:6]:
```suggestion
import type { TFunction } from 'i18next';
```

File: client/src/components/profile/profile.tsx [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import { Grid, Row } from '@freecodecamp/react-bootstrap';
 import React from 'react';
 import Helmet from 'react-helmet';
-import { TFunction, useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
```
Comment (Sboonny) [inline client/src/components/profile/profile.tsx:4]:
```suggestion
import type { TFunction } from 'i18next';
```

File: client/src/components/search/searchBar/search-bar.tsx [hunk 6]
```diff
@@ -1,7 +1,8 @@
 import { isEqual } from 'lodash-es';
 import React, { Component } from 'react';
 import { HotKeys, ObserveKeys } from 'react-hotkeys';
-import { TFunction, withTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
```
Comment (Sboonny) [inline client/src/components/search/searchBar/search-bar.tsx:4]:
```suggestion
import type { TFunction } from 'i18next';
```

**Review by Sboonny — COMMENTED** [review]
Import TFunction as type to reduce runtime and for readablity

**Review by naomi-lgbt — APPROVED** [review]

**Review by moT01 — APPROVED** [review]

**Pull request merged**